### DISCUSSION TITLE
How to use this in P2P  network building?

### DISCUSSION THREAD

**Discussion by DevSingh1101** [RingsNetwork/rings#476] [General]
Hi,

I am trying to make a P2P network for file sharing, how can I utilise this rings-node to make a P2P using rust? I am having trouble sending data to a different ip.

**Comment by RyanKung** [comment]
Hi, @gitsdestroyingmonster 

May I ask where you are in the process? From my understanding, you can use Rings DHT to implement file sharing, or you can simply use Rings to implement node handshaking and connection, and then implement file sharing on your own.

And, we have documents here, hope it helps: https://rings.gitbook.io/
